internal/inputs/generator_queries.go

```diff
@@ -195,67 +196,81 @@ func (g *QueryGenerator) init(config GeneratorConfig) error {
 	return nil
 }
 
+type factory struct {
+	database  string
+	generator interface{}
+}
+
 func (g *QueryGenerator) initFactories() error {
-	cassandra := &cassandra.BaseGenerator{}
```

Comment:
This refactoring must be sent in a separate pull request to upstream repository, since it is orthogonal to the rest of changes.